### ISSUE TITLE
Add language detection rule for locale-specific configuration

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#353]
## Summary

Add a content rule that detects the primary language of documentation and suggests appropriate language-specific configuration.

## Motivation

When documentation is written in a non-English language, there are often theme/configuration settings that should be adjusted:
- `theme.language` in mkdocs.yml (for MkDocs/Material)
- `language` in book.toml (for mdBook)
- Search index language settings
- Date/time formatting
- RTL support for Arabic, Hebrew, etc.

Currently users may not realize these settings exist or forget to configure them.

### Rule: CONTENT006 (or similar) - language-config-mismatch

1. Use [lingua-rs](https://github.com/pemistahl/lingua-rs) for language detection
2. Sample content from multiple markdown files to determine primary language
3. Compare detected language against configuration settings
4. Warn if mismatch detected

### Detection Strategy

- Sample first N paragraphs from each document
- Use lingua's confidence threshold to avoid false positives
- Consider code blocks should be excluded from detection
- Handle multilingual docs gracefully (maybe detect per-file?)

### Example Violations

```
CONTENT006: Documentation appears to be in German (de) but theme.language is set to 'en'
CONTENT006: Documentation appears to be in Japanese (ja), consider enabling CJK search support
CONTENT006: Documentation appears to be in Arabic (ar), consider enabling RTL support
```

### Configuration

```toml
[rules.CONTENT006]
# Minimum confidence for language detection (0.0-1.0)
confidence_threshold = 0.8
# Sample size (paragraphs per file)
sample_size = 5
# Languages to check for specific recommendations
check_rtl = true
check_cjk = true
```

## Dependencies

- [lingua-rs](https://crates.io/crates/lingua) - Accurate natural language detection

## Tasks

- [ ] Add lingua-rs dependency (feature-gated due to size)
- [ ] Implement language sampling from documents
- [ ] Create rule with configurable thresholds
- [ ] Add platform-specific recommendations (mdBook vs MkDocs)
- [ ] Add tests with multilingual content
- [ ] Document in rules-reference.md

## Notes

- lingua-rs is ~15MB due to language models - should be behind a feature flag
- Could also be useful for spell-checking integration later
- Consider lighter alternatives like whatlang-rs if size is a concern

**Comment by joshrotenberg** [comment]
Closing as stale/deferred. This has been sitting unactioned since January 2026 with no near-term plans, and the language detection idea is quite speculative. The MkDocs-specific angle also overlaps with the out-of-scope MkDocs support request. Can be reopened if it becomes a priority.
